pkg/cli/admin/mustgather/mustgather.go

```diff
@@ -414,7 +421,25 @@ func (o *MustGatherOptions) getGatherContainerLogs(pod *corev1.Pod) error {
 		ConsumeRequestFn: logs.DefaultConsumeRequest,
 		LogsForObject:    polymorphichelpers.LogsForObjectFn,
 		IOStreams:        genericclioptions.IOStreams{Out: newPrefixWriter(o.Out, fmt.Sprintf("[%s] POD", pod.Name))},
-	}).RunLogs()
+	}
+
+	for {
+		// gather script might take longer than the default API server time,
+		// so we should check if the gather script still runs and re-run logs
+		// thus we run this in a loop
+		if err := opts.RunLogs(); err != nil {
+			return err
+		}
+
+		klog.V(4).Infof("lost logs, re-trying...")
+
+		// to ensure we don't print all of history set since to past 2 seconds
+		opts.Options.(*corev1.PodLogOptions).SinceSeconds = &since2s
+
+		if done, _ := o.isGatherDone(pod); done {
```

Comment:
move this above the klog "lost logs, re-trying", so that only prints if !isGatherDone -
(see easily w/ `oc adm must-gather --dest-dir=/tmp/must-gather -v=4 -- date`)

Reply:
Good point, updated.